lighthouse-cli/performance-experiment/report/scripts/perf-x.js

```diff
@@ -31,7 +31,8 @@ class ConfigPanel {
     this._messageField = this._configPanel.querySelector('.js-message');
     this._urlBlockingList = this._configPanel.querySelector('.js-url-blocking-patterns');
     this._urlBlockingStatus = {};
-    this._reportId = new URL(window.location).searchParams.get('id');
+    const match = location.search.match(/[?&]id=([^&]*)/);
```

Comment:
We need to keep the deps for the viewer since it's a separate app. 

Since you've already done the work of moving over to the fallbacks, there's not a lot of benefit to adding polyfills for two small APIs. I'm fine with what you've got.